test/unit/concerns/annotating/model_test.rb

```diff
@@ -0,0 +1,150 @@
+          subject = AnnotatedFeature.new
+          subject.annotate(name, 'operator')
+
+          subject.expects(:remove_annotation).with(name)
+
+          subject.annotate(name, value)
+        end
+      end
+    end
+
+    class RemoveAnnotationTest < ModelTest
+      test 'marks annotation for destruction' do
+        name = 'managed_by'
+        subject = AnnotatedFeature.new
+        subject.annotate(name, 'operator')
+
+        subject.remove_annotation(name)
+
+        assert subject.annotations.first.marked_for_destruction?
+      end
+    end
+
+    class AnnotationTest < ModelTest
+      test 'returns the annotation model' do
+        name = 'managed_by'
+        subject = AnnotatedFeature.new
+        subject.annotate(name, 'operator')
+
+        result = subject.annotation(name)
+
+        assert_instance_of Annotation, result
+      end
+
+      test "returns nil if it doesn't exist" do
+        subject = AnnotatedFeature.new
+        subject.annotate('managed_by', 'operator')
+
+        result = subject.annotation('test')
+
+        assert_nil result
+      end
+    end
+
+    class ValueOfAnnotationTest < ModelTest
+      test 'returns the annotation value' do
+        name = 'managed_by'
+        value = 'operator'
+        subject = AnnotatedFeature.new
+        subject.annotate(name, value)
+
+        result = subject.value_of_annotation(name)
+
+        assert_equal value, result
+      end
+
+      test "returns nil if it doesn't exist" do
+        subject = AnnotatedFeature.new
+        subject.annotate('managed_by', 'operator')
+
+        result = subject.value_of_annotation('test')
+
+        assert_nil result
+      end
+    end
+
+    class AnnotationsTest < ModelTest
+      test 'extracts the proper parameters from the given hash' do
+        name = :managed_by
+        value = 'operator'
+        subject = AnnotatedFeature.new
+
+        subject.expects(:annotate).with(name, value)
+
+        subject.annotations = { "#{name}": value }
+      end
+
+      test 'annotates once per each received key' do
+        subject = AnnotatedFeature.new
+
+        subject.expects(:annotate).times(3)
+
+        subject.annotations = { managed_by: 'operator', something: 'else', foo: 'bar' }
+      end
+    end
+
+    class AnnotationsHash < ModelTest
+      test 'it properly builds a hash' do
+        name = 'managed_by'
+        value = 'operator'
+        subject = AnnotatedFeature.new
+        subject.annotate(name, value)
+        subject.save!
+
+        result = subject.reload.annotations_hash
+
+        assert_equal({ name => value }, result)
+      end
+    end
+
+    class AnnotationsXML < ModelTest
+      test 'it properly builds a xml' do
+        name = 'managed_by'
+        value = 'operator'
+        subject = AnnotatedFeature.new
+        subject.annotate(name, value)
+
+        result = subject.annotations_xml
+
+        assert_match "<annotations><#{name}>#{value}</#{name}></annotations>", result
```

Comment:
I'm not sure about the XML. Would it make sense to keep XML as:
```xml
<annotations>
  <annotation name="..." value="..."/>
</annotations>
```

or 
```xml
<annotations>
  <annotation name="...">value</annotation>
</annotations>
```
or whatever it generates by default? Because it's hard to validate elements with unspecified values... what probably XML users may do. idk, just current version doesn't look so clumsy as XML usually does. And probably doing things in a more expected way would be better.

I wonder if we have other examples in the code where we return an array of stuff and be consistent with it. For example tags :thinking:

Reply:
I'll check how it's done for other resources like tags. I'll try to be consistent.

Reply:
This is the XML representation for `extra_fields` and `notification_settings`:

```xml
<extra_fields>
  <cat_name>Paco</cat_name>
</extra_fields>
```
```xml
<notification_settings>
  <web_buyer>[0, 50, 80, 90, 100, 120, 150, 200, 300]</web_buyer>
  <email_buyer>[0, 50, 80, 90, 100, 120, 150, 200, 300]</email_buyer>
  <web_provider>[0, 50, 80, 90, 100, 120, 150, 200, 300]</web_provider>
  <email_provider>[0, 50, 80, 90, 100, 120, 150, 200, 300]</email_provider>
</notification_settings>
```

It's consistent with what we have now for annotations. But I can change it if you think that's better.

Reply:
That's great then, thank you for checking!